File: day-77 Algorithms & AI/homework/ex1.js

// orobiti warmoeba

function numB(num) {

  let machineNum = '';
  
  while (num > 0) {
    if (num % 2 === 0) {
      machineNum += '0';
    } else {
      machineNum += '1';
    }
    num = Math.floor(num / 2);
  }
  return machineNum || '0';
}

console.log(numB(10000));

File: day-077 Algorithms & AI/homework/ex1.py

# orobiti warmoeba

def numB(num):
    machineNum = ''
    
    while num > 0:
        if num % 2 == 0:
            machineNum += '0'
        else:
            machineNum += '1'
        num = num // 2
    
    return machineNum or '0'

print(numB(87))
